server-spi-private/src/main/java/org/keycloak/userprofile/UserProfileProvider.java

```diff
@@ -27,5 +29,13 @@
 
     UserProfileValidationResult validate(UserProfileContext updateContext, UserProfile updatedProfile);
 
-    boolean isReadOnlyAttribute(String key);
+    boolean isReadOnlyAttribute(UserUpdateEvent userUpdateEvent, String key);
```

Comment:
Why you need to rely on the `UserUpdateEvent` and not use the `UserProfileContext.ACCOUNT`. Similar to the signature of other methods in this interface?

Reply:
Thanks for your review and reply, but this can just be closed now, it is more than a year old and is not relevant anymore since all changes that happened after this was opened.